smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -619,6 +619,23 @@ contract MixinKeys is MixinErrors, MixinLockCore {
     );
   }
 
+  /**
+   * Set the expiration of a key
+   * @param _tokenId the id of the key
+   * @param _newExpiration the new timestamp to use
+   */
+  function setKeyExpiration(uint _tokenId, uint _newExpiration) public {
+    _onlyLockManager();
+    _keys[_tokenId].expirationTimestamp = _newExpiration;
```

Comment:
We probably do that in multiple places (expire, cancellation... etc)
I wonder if we should try to reuse some of the code?

Reply:
This seems similar to the issue identified in  #12553